src/utils/Celo-Integration/WakalaContractKit.ts

```diff
@@ -164,11 +182,49 @@ export default class WakalaContractKit {
     // }
   }
 
+  /**
+    * @dev Function to update a user's karma value for a specified application
+    * @param address The address of the user whose karma is being updated
+    * @param amount The amount used to calculate how much should be added or removed from
+                    a user's karma value. If the amount is positive, it will increase the 
+                    user's karma. If it is negative, it will decrease it. 
+    * @param updateFunctionKey An integer specifying which function should be used to update
+                               the user's karma. Setting the value to 1 will lead to a weighted 
+                               sum updated and setting it to 2 will lead to an averaged sum. 
+                               See README for details. => pass 2 for our case 
+                               karma contract link: https://github.com/karma-reputation-protocol/karma/tree/main
+
+    **/
+  updateKarma = async (address, amount, updateFunctionKey) => {
```

Comment:
You can do abstraction and inheritance to separate each smart contract method call.

ie WakalaContractKit extends KarmaContractKit extends WakalaBaseContractKit.

Then all the necessary variables gotten from the base as you go up.
Thanks.